packages/devtools_app/lib/src/ui/search.dart

```diff
@@ -1223,40 +1221,42 @@ class AutoCompleteController extends DisposableController
 class AutoCompleteMatch {
   AutoCompleteMatch(this.text, {this.matchedSegments});
 
-  final String text;
-  final List<Range> matchedSegments;
+  final String? text;
+  final List<Range>? matchedSegments;
 
   /// Transform the autocomplete match somehow (e.g. create a TextSpan where the
   /// matched segments are highlighted).
   T transformAutoCompleteMatch<T>({
-    T transformMatchedSegment(String segment),
-    T transformUnmatchedSegment(String segment),
-    T combineSegments(List<T> segments),
+    T transformMatchedSegment(String segment)?,
+    T transformUnmatchedSegment(String? segment)?,
+    T combineSegments(List<T> segments)?,
```

Comment:
make these all non-nullable and required. Some code smell to watch for as you're migrating: If something is unconditionally unwrapped where it is used, this is a good indication that it should probably be non-nullable.

Reply:
thanks. will watch more carefully for such cases